packages/react-compose/src/resolveSlotProps.ts

```diff
@@ -30,32 +31,25 @@ export function resolveSlotProps<TProps, TState = TProps>(
   Object.keys(slots).forEach((slotName: string) => {
     const slot = slots[slotName];
     // tslint:disable-next-line:no-any
-    let slotProp = (state as any)[slotName];
+    const slotProp = (state as any)[slotName];
 
     if (slot && slotProp !== undefined && slotProp !== null) {
-      const slotPropType = typeof slotProp;
-      const isLiteral = slotPropType === 'string' || slotPropType === 'number' || slotPropType === 'boolean';
+      const mergedSlotProp = mergeSlotProp(
+        slotProp,
+        slotProps[slotName],
+        (slot && slot.shorthandConfig && slot.shorthandConfig.mappedProp) || defaultMappedProps[slot],
+      );
 
-      // If the slot prop is a literal or JSX, pass it as a child of the slot.
-      if (isLiteral || React.isValidElement(slotProp)) {
-        const mappedProp =
-          (slot && slot.shorthandConfig && slot.shorthandConfig.mappedProp) || defaultMappedProps[slot] || 'children';
-        slotProp = { [mappedProp]: slotProp };
-      }
-
-      // If the children is a function, replace the slot.
-      if (typeof slotProp.children === 'function') {
+      if (typeof mergedSlotProp.children === 'function') {
         const { children, ...restProps } = slotProp;
-        slotProp.children = slotProp.children(slot, restProps);
-
+        // If the children is a function, replace the slot.
         slots[slotName] = React.Fragment;
+        slotProps[slotName] = {
+          children: slotProp.children({ Component: slot, props: { ...slotProps[slotName], ...restProps } }),
```

Comment:
This is not the same as previous and should be considered as a public API change.

Reply:
to clarify, `mergeProps` is currently only used by the V8 components we are working on, not by V0 compose yet.
And I'm trying to fix 2 things here:
1.  align the params interface for the render function interface with V0
2. we missed passing `...slotProps[slotName]` previously

Reply:
I updated the code slightly. If this is not the same as how V0 handles children render prop, pls let me know

Reply:
Yep, looks the same as in v0. Thanks 👍